File: src/main/java/fr/marstech/mtlinkspray/service/MailSenderServiceImpl.java
package fr.marstech.mtlinkspray.service;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.mail.SimpleMailMessage;
import org.springframework.mail.javamail.JavaMailSender;
import org.springframework.stereotype.Service;

@Service
public class MailSenderServiceImpl implements MailSenderService {

    private final JavaMailSender mailSender;

    @Value("${mt.link-spray.notification.mail.receiver:}")
    private String receiver;

    @Value("${mt.link-spray.notification.mail.sender:}")
    private String sender;

    @Autowired
    public MailSenderServiceImpl(JavaMailSender mailSender) {
        this.mailSender = mailSender;
    }

    @Override
    public void sendMail(String subject, String body) {
        SimpleMailMessage message = new SimpleMailMessage();
        message.setFrom(sender);
        message.setTo(receiver);
        message.setSubject(subject);
        message.setText(body);
        mailSender.send(message);
    }
}

File: src/main/kotlin/fr/marstech/mtlinkspray/service/MailSenderServiceImpl.kt
package fr.marstech.mtlinkspray.service

import org.springframework.beans.factory.annotation.Value
import org.springframework.mail.SimpleMailMessage
import org.springframework.mail.javamail.JavaMailSender
import org.springframework.stereotype.Service

@Service
class MailSenderServiceImpl(
    private val mailSender: JavaMailSender,
    @Value("\${mt.link-spray.notification.mail.receiver:}") private val receiver: String,
    @Value("\${mt.link-spray.notification.mail.sender:}") private val sender: String
) : MailSenderService {

    override fun sendMail(subject: String, body: String) {
        mailSender.send(SimpleMailMessage().apply {
            from = sender
            setTo(receiver)
            this.subject = subject
            text = body
        })
    }
}